x-pack/plugins/dashboard_mode/server/dashboard_mode_request_interceptor.js

```diff
@@ -25,9 +27,27 @@ export function createDashboardModeRequestInterceptor(dashboardViewerApp) {
     type: 'onPostAuth',
     async method(request, h) {
       const { auth, url } = request;
+      const user = auth.credentials;
+      const roles = user ? user.roles : [];
+
+      if (!user) {
```

Comment:
> Or the way it's done also related to these weird functional tests that depend on config object? If so, feel free to ignore this suggestion.

Yep, it was done this way because of the weird functional tests